File: truss-compiler/src/main/java/truss/compiler/support/ArgumentFlag.java
package truss.compiler.support;

public class ArgumentFlag extends Argument {
    public ArgumentFlag(String option, String description) {
        super(option, description, false);
    }

    @Override
    public boolean allowMultiple() {
        return false;
    }

    @Override
    public boolean hasParameter() {
        return false;
    }

    @Override
    public void setValue(String value) {
        throw new IllegalStateException();
    }
}

File: Truss.Compiler/Support/ArgumentFlag.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Support {
    public class ArgumentFlag : Argument {
        public ArgumentFlag(string option, string description)
            : base(option, description, false) {
        }

        public override bool AllowMultiple {
            get { return false; }
        }

        public override bool HasParameter {
            get { return false; }
        }

        public override void SetValue(string value) {
            throw new InvalidOperationException();
        }
    }
}
